xupload.py

```diff
@@ -0,0 +1,177 @@
+import aiohttp
+import aiofiles
+import asyncio
+import os
+
+
+class Xupload(object):
+
+    PROXY_SOCKET = ""
+    _QUERY_TIMEOUT = 60 * 60
+    _CHUNK_SIZE = 4 << 20
+
+    def __init__(self, upload_url, file_path, workers=1, headers={}, progress=None):
```

Comment:
Be careful when using mutable as default arguments, it'll be shared for all your `Xupload` initialisation (more info [here](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)):
```suggestion
    def __init__(self, upload_url, file_path, workers=1, headers=None, progress=None):
```

In your case, it's better to set it to `None` and check if its `None` in the body function and set it to an empty dictionary there.